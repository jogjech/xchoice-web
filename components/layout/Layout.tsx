import React, { FunctionComponent } from "react";
import { Layout as AntLayout } from "antd";
import Head from "../util/Head";
import Nav from "./Nav";
import Content from "./Content";
import Footer from "./Footer";
import styles from "./Layout.module.css";

interface Props {
  title: string;
  requiredAuth?: boolean;
}

const Layout: FunctionComponent<Props> = ({
  title,
  requiredAuth,
  children,
}) => (
  <AntLayout id={styles.siteLayout}>
    <Head title={title}></Head>
    <Nav></Nav>
    <Content requiredAuth={requiredAuth || false}>{children}</Content>
    <Footer></Footer>
  </AntLayout>
);

export default Layout;
