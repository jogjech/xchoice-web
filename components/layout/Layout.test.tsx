import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import Layout from "./Layout";

describe("Page layout", () => {
  const title = "Title";
  const children = <div id="#layout-children"></div>;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Layout title={title}>{children}</Layout>);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("contains head with title", () => {
    const head = wrapper.find("Head");
    expect(head.prop("title")).toEqual(title);
  });

  it("contains nav bar", () => {
    const nav = wrapper.find("Nav");
    expect(nav.exists()).toBeTruthy();
  });

  it("contains content", () => {
    const content = wrapper.find("Content");
    expect(content.exists()).toBeTruthy();
  });

  it("contains footer", () => {
    const footer = wrapper.find("Footer");
    expect(footer.exists()).toBeTruthy();
  });
});
