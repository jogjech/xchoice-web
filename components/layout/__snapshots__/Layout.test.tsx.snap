// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page layout renders correctly 1`] = `
<Layout
  id="siteLayout"
>
  <Head
    title="Title"
  />
  <Nav />
  <Content>
    <div
      id="#layout-children"
    />
  </Content>
  <Footer />
</Layout>
`;
